go-controller/pkg/node/ovspinning/ovspinning_linux.go

```diff
@@ -0,0 +1,185 @@
+//go:build linux
+// +build linux
+
+package ovspinning
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+	"golang.org/x/sys/unix"
+	"k8s.io/klog/v2"
+)
+
+// These variables are meant to be used in unit tests
+var tickDuration time.Duration = 10 * time.Second
+var getOvsVSwitchdPIDFn func() (string, error) = util.GetOvsVSwitchdPID
+var getOvsDBServerPIDFn func() (string, error) = util.GetOvsDBServerPID
+var featureEnablerFile string = "/etc/openvswitch/enable_dynamic_cpu_affinity"
+
+// Run monitors OVS daemon's processes (ovs-vswitchd and ovsdb-server) and sets their CPU affinity
+// masks to that of the current process.
+// This feature is enabled by the presence of a non-empty file in the path `/etc/openvswitch/enable_dynamic_cpu_affinity`
+func Run(stopCh <-chan struct{}) {
+
+	// The file must be present at startup to enable the feature
+	isFeatureEnabled, err := isFileNotEmpty(featureEnablerFile)
+	if err != nil {
+		klog.Warningf("Can't start OVS CPU affinity pinning: %v", err)
+		return
+	}
+
+	if !isFeatureEnabled {
+		klog.Info("OVS CPU affinity pinning disabled")
+		return
+	}
+
+	fileWatcher, err := createFileWatcherFor(featureEnablerFile)
+	if err != nil {
+		klog.Warningf("Can't start OVS CPU affinity pinning: %v", err)
```

Comment:
so if we get here we know that the user has enabled pinning, but we cant create a file watcher. Shouldn't this be an error at least?

Reply:
I think so. wdyt?

Reply:
Implemented a middle-ground solution: if we are not able to watch the file, the feature stays ON, the user can't stop the pinning on the fly (it needs a restart) and we avoid crash looping